.github/workflows/e2ePerformanceTests.yml

```diff
@@ -24,33 +24,21 @@ jobs:
     runs-on: ubuntu-latest
     name: Find the baseline and delta refs, and check for an existing build artifact for that commit
     outputs:
-      BASELINE_ARTIFACT_FOUND: ${{ steps.checkForExistingArtifact.outputs.ARTIFACT_FOUND }}
-      BASELINE_ARTIFACT_WORKFLOW_ID: ${{ steps.checkForExistingArtifact.outputs.ARTIFACT_WORKFLOW_ID }}
-      BASELINE_VERSION: ${{ steps.getMostRecentRelease.outputs.VERSION }}
+      BASELINE_REF: ${{ steps.getBaselineRef.outputs.BASELINE_REF }}
       DELTA_REF: ${{ steps.getDeltaRef.outputs.DELTA_REF }}
       IS_PR_MERGED: ${{ steps.getPullRequestDetails.outputs.IS_MERGED }}
     steps:
       - uses: actions/checkout@v4
         with:
-          # The OS_BOTIFY_COMMIT_TOKEN is a personal access token tied to osbotify (we need a PAT to access the artifact API)
-          token: ${{ secrets.OS_BOTIFY_COMMIT_TOKEN }}
-
-      - name: Get most recent release version
-        id: getMostRecentRelease
-        run: echo "VERSION=$(gh release list --limit 1 | awk '{ print $1 }')" >> "$GITHUB_OUTPUT"
-        env:
-          GITHUB_TOKEN: ${{ github.token }}
-
-      - name: Check if there's an existing artifact for this baseline
-        id: checkForExistingArtifact
-        uses: ./.github/actions/javascript/getArtifactInfo
-        with:
-          GITHUB_TOKEN: ${{ secrets.OS_BOTIFY_COMMIT_TOKEN }}
-          ARTIFACT_NAME: baseline-${{ steps.getMostRecentRelease.outputs.VERSION }}android-artifact-apk
+          fetch-depth: 0  # Fetches the entire history
```

Comment:
Does this work fine? the app history might be getting quite long

Reply:
@mountiny it was working fine. Maybe it was adding an overhead, but this operation was executed in ~1 min, so I think it's acceptable 🙂